Content.Server/Preferences/ServerPreferencesManager.cs

```diff
@@ -81,11 +82,12 @@ private PlayerPreferences GetFromYaml(string username)
             }
         }
 
-        // TODO: Check for characters that can't be in file names
-        // Might not be necessary depending on how we do authentication
         private static bool IsValidUsername(string username)
         {
-            return true;
+            if (username.All(char.IsLetterOrDigit))
+                return false;
+            var lowerUsername = username.ToLower();
+            return lowerUsername != "nul" && lowerUsername != "com"; // Reserved on windows
```

Comment:
1. Windows actually has quite a bit more reserved usernames than this. I actually have something checking for this in `ResourceManager` but that isn't enforced for userdata I don't think.

2. This entire function is kinda bad. Username validity should be enforced in `UsernameHelpers`. If you're gonna restrict it anywhere, do it there. That said I'd really rather not block usernames because of windows file system implementations, so maybe it'd be possible to store them as base64 or something on disk?

Reply:
What if we do away with usernames in filenames at all at this point? Just keep a big .yml with all the accounts? I expect we'll get to change this to use a real DB soon enough anyway

Reply:
Performance won't be great but once we move it to a DB that won't matter anyways, so go ahead.